ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx

```diff
@@ -11,242 +11,70 @@
-  const [kindToOptions, setKindToOptions] = useState<Record<string, unknown>>({ [defaultKind]: defaultOptions });
-  const options = kindToOptions[kind];
-
-  // TODO: What should we show if either of these error out?
-  const { data: plugin } = usePlugin('Panel', kind, {
-    enabled: kind !== '',
-    onSuccess: (plugin) => {
-      // Set initial options whenever plugin kind changes and a plugin is loaded
-      if (!kindToOptions[kind]) {
-        setKindToOptions({ ...kindToOptions, [kind]: plugin.createInitialOptions() });
-      }
-    },
-  });
-  const OptionsEditorComponent = plugin?.OptionsEditorComponent;
-
-  // TO DO: we might want to make the form a sub component we don't need this useEffect
-  // currently, we need to reset the states whenever panelDrawer is reopened
-  // since this component does not get remounted when it open/closes (otherwise we lose the animation of sliding in/out)
-  useEffect(() => {
-    setGroup(defaultGroup);
-    setPanelName(defaultPanelName);
-    setPanelDescription(defaultPanelDescription);
-    setKind(defaultKind);
-    setKindToOptions({ [defaultKind]: defaultOptions });
-  }, [defaultGroup, defaultPanelName, defaultPanelDescription, defaultKind, defaultOptions]);
-
-  const handleGroupChange: SelectProps<number>['onChange'] = (e) => {
-    const { value } = e.target;
-
-    // Handle string (which would be empty string but shouldn't happen since we don't allow a "None" option) by
-    // just ignoring it
-    if (typeof value === 'string') return;
-
-    setGroup(value);
-  };
-
-  const handlePanelNameChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setPanelName(e.target.value);
-  };
-
-  const handlePanelDescriptionChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setPanelDescription(e.target.value);
-  };
-
-  const handleKindChange: SelectProps<string>['onChange'] = (e) => {
-    setKind(e.target.value);
-  };
-
-  const handleOptionsChange: OptionsEditorProps<unknown>['onChange'] = (next) => {
-    setKindToOptions({ ...kindToOptions, [kind]: next });
-  };
-
-  const handleSubmit = (e: FormEvent) => {
-    e.preventDefault();
-
-    if (panelDrawer?.groupIndex !== undefined && !panelDrawer?.panelKey) {
-      addNewPanel();
-    } else if (panelDrawer?.panelKey) {
-      editPanel();
-    }
+import { useState } from 'react';
+import { Stack, Box, Button, Typography } from '@mui/material';
+import { Drawer } from '@perses-dev/components';
+import { useDashboardApp } from '../../context';
+import { usePanelDrawerModel } from './panel-editor-model';
+import { PanelEditorForm, panelEditorFormId, PanelEditorFormProps } from './PanelEditorForm';
+
+/**
+ * The Add/Edit panel drawer for editing a panel's options.
+ */
+export const PanelDrawer = () => {
+  const { closePanelDrawer } = useDashboardApp();
+  const model = usePanelDrawerModel();
+
+  // When the user clicks close, start closing but don't call the store yet to keep values stable during animtation
```

Comment:
it's not clear to me why we need this. Is it because you can somehow stop the drawer from closing?